TrackerApp/Onboarding/PageModel.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  OnboardingViewController.swift
+//  TrackerApp
+//
+//  Created by Юрий Демиденко on 30.04.2023.
+//
+
+import UIKit
+
+final class PageModel: UIViewController {
+
+    private var pageController: UIPageViewController?
+    private lazy var pages: [Pages] = Pages.allCases
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupPageController()
+        setupFinishOnboardingClosure()
+    }
+
+    init(pageController: UIPageViewController?) {
+        self.pageController = pageController
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupFinishOnboardingClosure() {
+        Pages.loginButtonTapAction = {
+            guard let window = UIApplication.shared.windows.first else {
+                fatalError("Invalid Configuration: unable to get window from UIApplication")
+            }
+            let tabBarController = TabBarController()
+            window.rootViewController = tabBarController
+            window.makeKeyAndVisible()
+        }
+    }
```

Comment:
как уже выше писала, переопредялять статику таким образом не стоит, это плохой паттерн, потому что я могу это сделать вообще откуда угодно и ничто мне не помешает это сделать. 

вполне ок, оставить здесь просто содержимое кложуры 
```suggestion
    private func setupFinishOnboarding() {
        guard let window = UIApplication.shared.windows.first else {
                fatalError("Invalid Configuration: unable to get window from UIApplication")
        }
        let tabBarController = TabBarController()
        window.rootViewController = tabBarController
        window.makeKeyAndVisible()
    }
```

и передать ее в инициализатор вью: 
```
PageView(with: pages[previousPageIndex], action: setupFinishOnboarding) 
``` 
здесь вместо кложуры { [weak self] in ... } сразу передаем ссылку на функцию `setupFinishOnboarding` (потребует указания сбегающего замыкания в инициализаторе @escaping)